### DISCUSSION TITLE
Welcome to SolarThing Discussions!

### DISCUSSION THREAD

**Discussion by retrodaredevil** [wildmountainfarms/solarthing#189] [Announcements]
## 👋 Welcome!

Hello! I am enabling GitHub Discussions on the SolarThing repository! This will hopefully encourage more people to interact with the SolarThing community and encourage more feedback about SolarThing.

Issues will still be used heavily, and if you have an issue, feature request, or something else, you can (and should) use the issues page. I hope that by opening up discussions here I'll here from more people about their experience with SolarThing.

**Comment by orlandoinsane** [comment]
Hi just discovered your project and it looks amazing. I have a bunch of Renogy stuff that basically doesn’t work using their own proprietary gear, so hoping solarthings might be a solution. 

I have a Renogy Lycan 5000, which is really just a hodge podge of their 48v hybrid charger and inverter, 48v batteries and some breaker/connectors in a heavy box. I know you currently don’t support inverters but am curious if there’s any data that can be pulled from the unit via rs485.

also it appears that Renogy no longer sells their RS485 to usb connector so the fact that you have the doc with the pinout is awesome. 

I’m a bit of a noob with coding but am very hardware oriented so hoping I can make something work!

**Reply by retrodaredevil** [reply]
Hi @orlandoinsane! You can definitely pull data from the inverter. My understanding is that all Renogy inverters use the Modbus protocol, but the registers the inverters support are different.

Usually the common register between Renogy products is the battery voltage register, along with a couple of other common ones. There's a good chance that if you were to use `solarthing check`, it would read your battery voltage, but of course SolarThing doesn't support inverters at this time, so beyond `solarthing check`, SolarThing cannot help you.

My recommendation is to first confirm that communication is working between your inverter and your Raspberry Pi or computer you are using to communicate with this. Easiest way to do this is to use`solarthing check` via docker (https://solarthing.readthedocs.io/en/latest/quickstart/serial-port/solarthing-check.html) because it's something I know works and is a tool that gives decent error messages. It does require you to either install Docker, or go through the legacy native install instructions (I recommend the docker installation, as it leaves behind the least amount of mess).

If you want to just pull data off of it, I would recommend looking at [solarshed code](https://github.com/corbinbs/solarshed/blob/master/solarshed/controllers/renogy_rover.py) because I think it's one of the easiest Python examples to understand. I don't really like solarshed's heavy use of inheritance, but it's still a pretty simple example.

The Modbus registers for Renogy inverters is documented somewhere out there, but I'm not really sure where to find the PDF specifically for the inverter. You can find the PDF for the charge controller here: https://diysolarforum.com/threads/renogy-rs-485-communications-interface-in-detail.12495/ or here: https://github.com/wildmountainfarms/solarthing/tree/master/other/solar/resources

I tried emailing Renogy a few years back about obtaining a document documenting the inverter modbus registers, but they wanted me to sign an NDA, and I'm not sure if what I wanted to do with that document would violate the NDA, so I dropped it.

**Reply by orlandoinsane** [reply]
Cool, I’ll take a look at this! 

I am very much a novice at anything code related… but got to start somewhere! I could consider signing that NDA, I wouldn’t be surprised if the documentation leaked somehow anonymously on a site for you then be able to read them. 😉

they do have an email address they gave me re API access, maybe I could inquire there…

**Reply by retrodaredevil** [reply]
There is an issue related to this: https://github.com/wildmountainfarms/solarthing/issues/137

So feel free to keep up with that issue. TLDR is that I want to support parts of these inverters (datapoints such as battery voltage at least), but I don't have a timeline on when or if I plan to work on it.

**Comment by djuniversal** [comment]
Maybe I'm the only one with this problem. I have searched for any further information for a while but am coming up empty. I have an outback system outputting from a mate 2 serial to a raspberry pi 4. I cannot get the daily accumulated graphs in grafana to carry until midnight. They cut off at 17:00 each day. I am in US Pacific time so I guess that's them cutting off at GMT. I have had a cursory look back through the stack and it seems like it is how the data is recorded in the database (using couchdb).

I am assuming that you calculate this on the serial data and store it as a variable rather than it being a value that is outputted from the mate because the mate display keeps calculating the value and cuts over at the correct time.

There was a note in your support docs to making sure the raspberry pi has the time zone set correctly but no further notes. I believe I've got it set up right, the raspberry pi, the mate, and grafana are all set to the same time zone (US Pacific).

Any ideas?

**Reply by retrodaredevil** [reply]
My guess is that you have a docker install and that the timezone within the docker container is not set correctly. 

You can simply add a `TZ=America/Los_Angeles` in your docker compose's list of environment variables, or if you would prefer to, you can get the timezones in sync by adding the lines shown here: https://stackoverflow.com/a/44440563

I recommend setting this for all SolarThing docker containers, but the one that actually determines when the daily cutoff is is the container running SolarThing server, which hosts all the GraphQL data. 

I'll take a look at the docs and see if I can make this more clear for docker installs. Let me know if this fixes your problem.

**Reply by retrodaredevil** [reply]
As an additional comment, you note that SolarThing stores the data within the database just like the MATE outputs data. If you were to graph that raw value, it usually resets in the morning around 6AM (at least for me it does). For older MX devices (or maybe older MATE devices, I'm not sure) that reset time is based on the sunlight that the controller is getting. I think the MATE has logic to reset if it's gone a night without charging and then it resets when the sun comes back out. 

Anyway, I decided to calculate that field because not only would my MX devices reset at slightly different times, but they would reset at different times than my other charge controllers. So having them reset at midnight (in the local timezone) makes the most sense.

**Reply by djuniversal** [reply]
Great, thank you. I have added that to each container (I looked for a way to do a variable for all containers and could only find that it was a limitation of docker). I will let you know later today if it works. Thank you for the prompt reply.

Also, as a side note this has been an very helpful program for us. We are a family of 9 visiting my in-laws (only two of them). They are completely off-grid (solar + generator backup) and we knew his system was large enough but we had a drop-out early one morning during our visit. This system has helped us diagnose issues in his solar setup and make some key changes/fixes so that they can easily have family visits without any trouble in future. I really appreciate you making this and putting it up for all to use.

**Reply by djuniversal** [reply]
Well, that worked perfectly.
I used the method from the stack overflow article and applied it to all my containers. Apparently that method isn't good if you have a docker instance on a cloud server (then you would use the environment variable as above) but for local server it works no problem.

**Reply by retrodaredevil** [reply]
Glad you got it working. I'm always happy to hear about how others are using SolarThing!

**Comment by CapnJackOff** [comment]
Well, I can't believe this MPPT controller still works after 5 years, but it does, and of course Solarthing is still doing it's thing like a champ. I have a lot of friends using solar and I have toyed with the idea of just cloning this RPi running Solarthing, InfluxDB and Grafana and having my friends plug it in and go. The weak link has been of course the cable. I have struggled to find a consistent plug and play solution since the beginning. Now, I seem to have found a source for the cables. A seller on AliExpress is selling the RichSOLAR cables, and has at this time more than 2500 of them. Of course the RichSOLAR MPPT is made by SRNE, maker of Renogy Rover, Grape Solar Zenith and probably others.
I ordered 5 and they are legit - work perfectly.

Here is a link to the store item.
https://www.aliexpress.us/item/3256806131726777.html

**Comment by retrodaredevil** [comment]
I welcome anyone to post on SolarThing Discussions. From here on out, please create your own discussion in [General](https://github.com/wildmountainfarms/solarthing/discussions/categories/general).
